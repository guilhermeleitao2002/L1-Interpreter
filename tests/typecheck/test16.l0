type Circle = struct { #cx:int, #cy:int, #rad:int };
type Rectangle = struct { #h:int, #w:int };
type Blob = union { #circle: Circle, #rect: Rectangle };
let r1:Blob = #rect({ #h = 10, #v = 20 });
match r1 {
        #circle(c) -> let r = pi*c.#rad; (r*r)/2
    |   #rect(r) -> r.#h * r.#v
}
;;