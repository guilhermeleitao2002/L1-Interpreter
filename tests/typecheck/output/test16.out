TypeError: Expression type union { rect: struct { v: int, h: int } } does not match declared type union { rect: Rectangle, circle: Circle }